[backport 2025.2]sstables_loader: Fix load-and-stream vs skip-cleanup check

regevran opened the pull request:
The intention was to fail the REST API call in case --skip-cleanup is requested for --load-and-stream loading. The corresponding if expression is checking something else :( despite log message is correct.

Fixes: https://github.com/scylladb/scylladb/issues/24913

(cherry picked from commit f14e1c1a8a071d434bba28da3f16ba32b9a52a01)

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/202/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/202/)
:white_check_mark: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release/269/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release-with-tablets/255/)
:white_check_mark: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release-with-consistent-topology-changes/265/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/202/)
#### Build Details:
- Duration: 7 hr 18 min
- Builder: i-049eae5ed90a7142f (m5d.16xlarge)

denesb commented:
> (cherry picked from commit [4cf7042](https://github.com/scylladb/scylladb/commit/4cf7042de7c99a886e6ae1c862be43044ae27149))

"This commit does not belong to any branch on this repository, and may belong to a fork outside of the repository."

Where did you cherry-pick this commit from?
Is this from a PR?

regevran commented:
This PR is so simple, just remove a not (!).
But it got complicated due to 1) automation, 2) my mistake.

1) The automation (scylladbbot) created the original PR based on a label I added. But, it had conflicts which I fixed but could not force push (to remove the committed conflicts by the scylladbbot)

2) I closed the auto created PR and created a manual one - which mistakenly I tried to merge to master instead of to 2025.2.

3) I closed the second PR, created a third PR, that probably referenced the second PR commit instead of the original one I backported from.

4) Now I fixed it, and this PR I _think_ is ok now. @denesb - can you please take a second look?

If you are really into looking at the details of all this, see GH reports (i.e. the lines being added when actions are happening elsewhere) in the issue starting [here](https://github.com/scylladb/scylladb/issues/24913#issuecomment-3066963915).

denesb commented:
I queued this PR. The cherry-pick marker is still wrong, but I tracked down the original commit (https://github.com/scylladb/scylladb/commit/bd3bd089e1754f95b37bdfb05681bf1376aca746) and fixed it when I queued it.

scylladb-promoter commented:
#### :green_circle: CI State: SUCCESS

:white_check_mark: - [Framework test](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/218/)
:white_check_mark: - [Build](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/218/)
:white_check_mark: - [dtest with tablets](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release-with-tablets/270/)
:white_check_mark: - [dtest with consistent topology changes](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release-with-consistent-topology-changes/279/)
:white_check_mark: - [dtest with gossip topology changes](https://jenkins.scylladb.com/job/scylla-2025.2/job/gating-dtest-release/284/)
:white_check_mark: - [Unit Tests](https://jenkins.scylladb.com/job/scylla-2025.2/job/scylla-ci/218/)
#### Build Details:
- Duration: 7 hr 23 min
- Builder: i-0e2c31f2e28254307 (m5d.16xlarge)

mykaul commented:
https://github.com/scylladb/scylladb/commit/71e9f5e662b0416b2615f4a4e3ac7670ec14052d
